website/sales/models/shift.py

```diff
@@ -0,0 +1,189 @@
+            """(SELECT CAST(COALESCE(SUM("__orders"."total__"), 0) AS NUMERIC) AS "shift_revenue__"
+                FROM (
+                    SELECT "sales_order"."id", "sales_order"."shift_id", "sales_order"."discount", "sales_order"."payment_id", CAST(SUM("sales_orderitem"."total") AS NUMERIC) AS "subtotal__", CAST((SUM("sales_orderitem"."total") - COALESCE("sales_order"."discount", 0)) AS NUMERIC) AS "total__", SUM("sales_orderitem"."amount") AS "num_items__"
+                    FROM "sales_order" LEFT JOIN "sales_orderitem" ON "sales_orderitem"."order_id" = "sales_order"."id"
+                    GROUP BY "sales_order"."id", "sales_order"."shift_id", "sales_order"."discount"
+                    ) AS "__orders"
+                WHERE "__orders"."shift_id"="sales_shift"."id"
+                AND ("__orders"."payment_id" IS NOT NULL OR ("__orders"."payment_id" IS NULL AND "__orders"."total__"=0))
+                )""",
+            [],
+        )
+
+    @queryable_property(annotation_based=True)
+    @classmethod
+    def num_orders(cls):
+        return Count("orders")
+
+    @queryable_property(annotation_based=True)
+    @classmethod
+    def num_orders_paid(cls):
+        return RawSQL(
+            """(SELECT COUNT(*) AS "num_orders__"
+                FROM (
+                    SELECT "sales_order"."id", "sales_order"."shift_id", "sales_order"."discount", "sales_order"."payment_id", CAST(SUM("sales_orderitem"."total") AS NUMERIC) AS "subtotal__", CAST((SUM("sales_orderitem"."total") - COALESCE("sales_order"."discount", 0)) AS NUMERIC) AS "total__", SUM("sales_orderitem"."amount") AS "num_items__"
+                    FROM "sales_order" LEFT JOIN "sales_orderitem" ON "sales_orderitem"."order_id" = "sales_order"."id"
+                    GROUP BY "sales_order"."id", "sales_order"."shift_id", "sales_order"."discount"
+                    ) AS "__orders"
+                WHERE "__orders"."shift_id"="sales_shift"."id"
+                AND ("__orders"."payment_id" IS NOT NULL OR ("__orders"."payment_id" IS NULL AND "__orders"."total__"=0))
+                )""",
+            [],
+        )
+
+    @property
+    def product_sales(self):
+        qs = (
+            self.orders.exclude(order_items__isnull=True)
+            .values("order_items__product")
+            .distinct()
+            .annotate(sold=Sum("order_items__amount"))
+        )
+        return {
+            item[0]: item[1]
+            for item in qs.values_list("order_items__product__product__name", "sold")
+        }
+
+    @property
+    def title(self):
+        if self.custom_title:
+            return self.custom_title
+        if hasattr(self, "event"):
+            return self.event.title
+        return ""
+
+    def __str__(self):
+        if self.title != "":
+            return f"Shift {self.pk} - {self.title}"
+        return f"Shift {self.pk}"
+
+    @staticmethod
+    def currently_active():
+        return Shift.objects.filter(active=True)
```

Comment:
I think a better way to do this would be creating a custom manager and making it `Shift.active_objects`

Reply:
I rather don't because then I would need to override the QueryablePropertiesManager for that. But this code isn't used anymore anyway, so I'll just remove it